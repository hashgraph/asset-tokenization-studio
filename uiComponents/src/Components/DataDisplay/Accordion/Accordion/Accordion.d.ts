import type { AccordionProps as ChakraAccordionProps } from "@chakra-ui/react";
import type { SystemStyleObject as ChakraSystemStyleObject, ComponentWithAs } from "@chakra-ui/system";
import React from "react";
import type { BaseMultiStyleConfiguration, PartsStyleInterpolation } from "@/Theme/Components/BaseMultiStyleConfiguration";
export interface AccordionProps extends Omit<ChakraAccordionProps, "children" | "title" | "description"> {
    children: React.ReactNode;
    description?: string;
    title: string;
}
declare const accordionPartsList: Array<string>;
type Parts = typeof accordionPartsList;
export interface AccordionThemeConfiguration extends BaseMultiStyleConfiguration<Parts> {
    baseStyle: (() => Partial<AccordionThemeStyle>) | PartsStyleInterpolation<Parts>;
}
export type AccordionThemeStyle = Record<Parts[number], ChakraSystemStyleObject>;
export declare const Accordion: ComponentWithAs<"div", AccordionProps>;
export {};
