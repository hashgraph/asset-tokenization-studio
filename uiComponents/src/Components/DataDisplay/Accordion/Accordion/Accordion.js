var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { Accordion as ChakraAccordion } from "@chakra-ui/accordion";
import { Box as ChakraBox, Divider as ChakraDivider } from "@chakra-ui/layout";
import { forwardRef, useMultiStyleConfig as useChakraMultiStyleConfig, } from "@chakra-ui/system";
import React from "react";
import { Text } from "@/Components/Foundations/Text";
var accordionPartsList = ["container", "title", "item"];
export var Accordion = forwardRef(function (_a, ref) {
    var children = _a.children, description = _a.description, title = _a.title, props = __rest(_a, ["children", "description", "title"]);
    var styles = useChakraMultiStyleConfig("Accordion");
    return (React.createElement(ChakraAccordion, __assign({ ref: ref }, props, { sx: styles.container }),
        React.createElement(ChakraBox, { sx: styles.titleContainer },
            React.createElement(Text, { sx: styles.title }, title)),
        React.createElement(ChakraDivider, { sx: styles.divider }),
        description && (React.createElement(ChakraBox, { sx: styles.descriptionContainer },
            React.createElement(Text, { sx: styles.description }, description))),
        React.createElement(ChakraBox, { sx: styles.itemsContainer }, children)));
});
