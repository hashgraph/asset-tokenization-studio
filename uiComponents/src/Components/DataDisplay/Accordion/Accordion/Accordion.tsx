import type { AccordionProps as ChakraAccordionProps } from "@chakra-ui/react";
import { Accordion as ChakraAccordion } from "@chakra-ui/accordion";
import { Box as ChakraBox, Divider as ChakraDivider } from "@chakra-ui/layout";
import type {
  SystemStyleObject as ChakraSystemStyleObject,
  ComponentWithAs,
} from "@chakra-ui/system";
import {
  forwardRef,
  useMultiStyleConfig as useChakraMultiStyleConfig,
} from "@chakra-ui/system";
import React from "react";
import type {
  BaseMultiStyleConfiguration,
  PartsStyleInterpolation,
} from "@/Theme/Components/BaseMultiStyleConfiguration";
import { Text } from "@/Components/Foundations/Text";

export interface AccordionProps
  extends Omit<ChakraAccordionProps, "children" | "title" | "description"> {
  children: React.ReactNode;
  description?: string;
  title: string;
}

const accordionPartsList: Array<string> = ["container", "title", "item"];

type Parts = typeof accordionPartsList;

export interface AccordionThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {
  baseStyle:
    | (() => Partial<AccordionThemeStyle>)
    | PartsStyleInterpolation<Parts>;
}

export type AccordionThemeStyle = Record<
  Parts[number],
  ChakraSystemStyleObject
>;

export const Accordion: ComponentWithAs<"div", AccordionProps> = forwardRef<
  AccordionProps,
  "div"
>(({ children, description, title, ...props }: AccordionProps, ref) => {
  const styles = useChakraMultiStyleConfig("Accordion");

  return (
    <ChakraAccordion ref={ref} {...props} sx={styles.container}>
      <ChakraBox sx={styles.titleContainer}>
        <Text sx={styles.title}>{title}</Text>
      </ChakraBox>
      <ChakraDivider sx={styles.divider} />
      {description && (
        <ChakraBox sx={styles.descriptionContainer}>
          <Text sx={styles.description}>{description}</Text>
        </ChakraBox>
      )}
      <ChakraBox sx={styles.itemsContainer}>{children}</ChakraBox>
    </ChakraAccordion>
  );
});
