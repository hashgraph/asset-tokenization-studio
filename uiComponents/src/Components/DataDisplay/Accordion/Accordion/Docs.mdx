import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as AccordionStories from "./Accordion.stories";

<Meta of={AccordionStories} />

<Title />

<Primary />

## How to set up

### Config theme

Add the following configurations to the `theme.tsx`, `Accordion.ts` and `AccordionItem.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    Accordion: ConfigAccordion,
    AccordionItem: ConfigAccordionItem,
  },
};
```

```tsx dark
const ConfigAccordion: AccordionThemeConfiguration = {
  parts: accordionPartsList,
  baseStyle: {},
};
```

```tsx dark
const ConfigAccordionItem: AccordionItemThemeConfiguration = {
  parts: accordionItemPartsList,
  baseStyle: {},
};
```

### Component parts

You have these parts available:

- `"container", "title", "item"` (`accordionPartsList`)
- `"button", "item", "panel"` (`accordionItemPartsList`)

## Props

<ArgTypes of={AccordionStories} />

## Stories

<Stories title="" />
