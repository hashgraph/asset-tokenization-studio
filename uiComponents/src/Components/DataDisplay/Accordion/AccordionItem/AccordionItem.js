var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { AccordionIcon as ChakraAccordionIcon, AccordionItem as ChakraAccordionItem, AccordionButton as ChakraAccordionButton, AccordionPanel as ChakraAccordionPanel, } from "@chakra-ui/react";
import { Box as ChakraBox } from "@chakra-ui/layout";
import { forwardRef, useMultiStyleConfig as useChakraMultiStyleConfig, } from "@chakra-ui/system";
import React from "react";
var accordionItemPartsList = ["button", "item", "panel"];
export var AccordionItem = forwardRef(function (_a, ref) {
    var children = _a.children, customTitle = _a.customTitle, title = _a.title, icon = _a.icon, props = __rest(_a, ["children", "customTitle", "title", "icon"]);
    var styles = useChakraMultiStyleConfig("AccordionItem");
    return (React.createElement(ChakraAccordionItem, __assign({ ref: ref }, props, { sx: styles.container }),
        title && (React.createElement("h2", null,
            React.createElement(ChakraAccordionButton, { sx: styles.button },
                React.createElement(ChakraBox, { sx: styles.title }, title),
                React.createElement(ChakraAccordionIcon, __assign({}, (icon && { as: icon })))))),
        customTitle && React.createElement(ChakraBox, { w: "full" }, customTitle),
        React.createElement(ChakraAccordionPanel, { sx: styles.panel }, children)));
});
