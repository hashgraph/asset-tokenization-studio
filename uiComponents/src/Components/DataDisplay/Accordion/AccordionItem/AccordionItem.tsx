import type { AccordionItemProps as ChakraAccordionItemProps } from "@chakra-ui/react";
import {
  AccordionIcon as ChakraAccordionIcon,
  AccordionItem as ChakraAccordionItem,
  AccordionButton as ChakraAccordionButton,
  AccordionPanel as ChakraAccordionPanel,
} from "@chakra-ui/react";
import { Box as ChakraBox } from "@chakra-ui/layout";
import type {
  SystemStyleObject as ChakraSystemStyleObject,
  ComponentWithAs,
} from "@chakra-ui/system";
import {
  forwardRef,
  useMultiStyleConfig as useChakraMultiStyleConfig,
} from "@chakra-ui/system";
import React from "react";
import type {
  BaseMultiStyleConfiguration,
  PartsStyleInterpolation,
} from "@/Theme/Components/BaseMultiStyleConfiguration";

export interface AccordionItemProps
  extends Omit<ChakraAccordionItemProps, "children" | "title"> {
  children: React.ReactNode;
  title?: React.ReactNode;
  icon?: any;
  customTitle?: React.ReactNode;
}

const accordionItemPartsList: Array<string> = ["button", "item", "panel"];

type Parts = typeof accordionItemPartsList;

export interface AccordionItemThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {
  baseStyle:
    | (() => Partial<AccordionItemThemeStyle>)
    | PartsStyleInterpolation<Parts>;
}

type AccordionItemThemeStyle = Record<Parts[number], ChakraSystemStyleObject>;

export const AccordionItem: ComponentWithAs<"div", AccordionItemProps> =
  forwardRef<AccordionItemProps, "div">(
    (
      { children, customTitle, title, icon, ...props }: AccordionItemProps,
      ref
    ) => {
      const styles = useChakraMultiStyleConfig("AccordionItem");

      return (
        <ChakraAccordionItem ref={ref} {...props} sx={styles.container}>
          {title && (
            <h2>
              <ChakraAccordionButton sx={styles.button}>
                <ChakraBox sx={styles.title}>{title}</ChakraBox>
                <ChakraAccordionIcon {...(icon && { as: icon })} />
              </ChakraAccordionButton>
            </h2>
          )}
          {customTitle && <ChakraBox w="full">{customTitle}</ChakraBox>}
          <ChakraAccordionPanel sx={styles.panel}>
            {children}
          </ChakraAccordionPanel>
        </ChakraAccordionItem>
      );
    }
  );
