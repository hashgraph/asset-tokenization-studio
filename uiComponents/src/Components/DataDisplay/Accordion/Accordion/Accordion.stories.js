var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { Accordion } from ".";
import { AccordionItem } from "../AccordionItem";
import { Box as ChakraBox } from "@chakra-ui/layout";
var AccordionItems = function () {
    return (React.createElement(ChakraBox, null,
        React.createElement(AccordionItem, { title: "First Element" },
            React.createElement("p", null, "Content in First Element")),
        React.createElement(AccordionItem, { title: "Second Element" },
            React.createElement("p", null, "Content in Second Element")),
        React.createElement(AccordionItem, { title: "Third Element" },
            React.createElement("p", null, "Content in Third Element"))));
};
var meta = {
    title: "Design System/Data Display/Accordion",
    component: Accordion,
    argTypes: {},
    parameters: {
        design: {
            type: "figma"
        },
        docs: {}
    },
    args: {
        children: React.createElement(AccordionItems, null)
    }
};
export default meta;
var Template = function (args) {
    return React.createElement(Accordion, __assign({}, args));
};
export var Default = Template.bind({});
Default.args = {
    title: "Default Accordion"
};
export var DefaultWithDescription = Template.bind({});
DefaultWithDescription.args = {
    title: "Accordion with description",
    description: "By default when open an item any other expanded item may be collapsed again"
};
export var AllowMultiple = Template.bind({});
AllowMultiple.args = {
    title: "Expand multiple items at once",
    description: "The accordion permit multiple items to be expanded at once",
    allowMultiple: true
};
