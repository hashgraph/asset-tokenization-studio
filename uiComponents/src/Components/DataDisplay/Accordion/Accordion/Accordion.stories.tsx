import type { Meta, StoryFn } from "@storybook/react";
import React from "react";
import { Accordion } from ".";
import { AccordionItem } from "../AccordionItem";
import { Box as ChakraBox } from "@chakra-ui/layout";

const AccordionItems = () => {
  return (
    <ChakraBox>
      <AccordionItem title="First Element">
        <p>Content in First Element</p>
      </AccordionItem>
      <AccordionItem title="Second Element">
        <p>Content in Second Element</p>
      </AccordionItem>
      <AccordionItem title="Third Element">
        <p>Content in Third Element</p>
      </AccordionItem>
    </ChakraBox>
  );
};

const meta = {
  title: "Design System/Data Display/Accordion",
  component: Accordion,
  argTypes: {},
  parameters: {
    design: {
      type: "figma",
    },
    docs: {},
  },
  args: {
    children: <AccordionItems />,
  },
} as Meta<typeof Accordion>;

export default meta;

const Template: StoryFn<typeof Accordion> = (args) => {
  return <Accordion {...args} />;
};

export const Default = Template.bind({});
Default.args = {
  title: "Default Accordion",
};

export const DefaultWithDescription = Template.bind({});
DefaultWithDescription.args = {
  title: "Accordion with description",
  description:
    "By default when open an item any other expanded item may be collapsed again",
};

export const AllowMultiple = Template.bind({});
AllowMultiple.args = {
  title: "Expand multiple items at once",
  description: "The accordion permit multiple items to be expanded at once",
  allowMultiple: true,
};
