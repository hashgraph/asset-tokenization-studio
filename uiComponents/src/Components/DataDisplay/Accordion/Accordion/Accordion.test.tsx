import { render } from "@/test-utils";
import { Box, Text } from "@chakra-ui/react";
import type { RenderResult } from "@testing-library/react";
import { waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import type { AccordionProps } from "./Accordion";
import { Accordion } from "./Accordion";
import { AccordionItem } from "../AccordionItem";

const title = "Accordion Title";
const description = "Accordion Description";
const items = [
  {
    title: "First Element",
    content: "Content in First Element",
  },
  {
    title: "Second Element",
    content: "Content in Second Element",
  },
  {
    title: "Third Element",
    content: "Content in Third Element",
  },
];
const AccordionItems = () => {
  return (
    <Box>
      <AccordionItem title={items[0].title}>
        <Text>{items[0].content}</Text>
      </AccordionItem>
      <AccordionItem title={items[1].title}>
        <Text>{items[1].content}</Text>
      </AccordionItem>
      <AccordionItem title={items[2].title}>
        <Text>{items[2].content}</Text>
      </AccordionItem>
    </Box>
  );
};

const defaultProps: AccordionProps = {
  title,
  children: <AccordionItems />,
};

describe(`<Accordion />`, () => {
  const factoryComponent = (props: AccordionProps = defaultProps) =>
    render(<Accordion {...props} />);

  const checkAllIsCollapsed = (component: RenderResult) => {
    items.forEach((item) => {
      expect(component.getByText(item.title)).toBeVisible();
      expect(component.queryByText(item.content)).not.toBeVisible();
    });
  };

  test("should render correctly", () => {
    const component = factoryComponent();

    expect(component).toMatchSnapshot();
  });

  test("should render with description", () => {
    const component = factoryComponent({ ...defaultProps, description });

    expect(component.getByText(description)).toBeVisible();
  });

  test("should render collapsed items", () => {
    const component = factoryComponent();

    checkAllIsCollapsed(component);
  });

  // TODO: improve this test, works in local but fails in pipeline
  test.skip("by default when open an item all other items are collapsed", async () => {
    const component = factoryComponent({
      ...defaultProps,
    });

    checkAllIsCollapsed(component);

    userEvent.click(component.getByText(items[0].title));

    await waitFor(() => {
      expect(component.getByText(items[0].content)).toBeVisible();
      expect(component.queryByText(items[1].content)).not.toBeVisible();
      expect(component.queryByText(items[2].content)).not.toBeVisible();
    });

    userEvent.click(component.getByText(items[1].title));

    await waitFor(() => {
      expect(component.queryByText(items[0].content)).not.toBeVisible();
      expect(component.getByText(items[1].content)).toBeVisible();
      expect(component.queryByText(items[2].content)).not.toBeVisible();
    });

    userEvent.click(component.getByText(items[2].title));

    await waitFor(() => {
      expect(component.queryByText(items[0].content)).not.toBeVisible();
      expect(component.queryByText(items[1].content)).not.toBeVisible();
      expect(component.getByText(items[2].content)).toBeVisible();
    });
  });

  test.skip("with AllowMultiple all items could be expanded", async () => {
    const component = factoryComponent({
      ...defaultProps,
      allowMultiple: true,
    });

    checkAllIsCollapsed(component);

    userEvent.click(component.getByText(items[0].title));

    await waitFor(() => {
      expect(component.getByText(items[0].content)).toBeVisible();
      expect(component.queryByText(items[1].content)).not.toBeVisible();
      expect(component.queryByText(items[2].content)).not.toBeVisible();
    });

    userEvent.click(component.getByText(items[1].title));

    await waitFor(() => {
      expect(component.getByText(items[0].content)).toBeVisible();
      expect(component.getByText(items[1].content)).toBeVisible();
      expect(component.queryByText(items[2].content)).not.toBeVisible();
    });

    userEvent.click(component.getByText(items[2].title));

    await waitFor(() => {
      expect(component.getByText(items[0].content)).toBeVisible();
      expect(component.getByText(items[1].content)).toBeVisible();
      expect(component.getByText(items[2].content)).toBeVisible();
    });
  });
});
