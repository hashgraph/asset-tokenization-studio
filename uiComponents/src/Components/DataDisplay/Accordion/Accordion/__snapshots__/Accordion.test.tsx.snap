// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div>
      <div
        class="chakra-accordion css-1wbdjbz"
      >
        <div
          class="css-0"
        >
          <p
            class="css-0"
          >
            Accordion Title
          </p>
        </div>
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-svjswr"
        />
        <div
          class="css-0"
        >
          <div
            class="css-0"
          >
            <div
              class="chakra-accordion__item css-17mg6aq"
            >
              <h2>
                <button
                  aria-controls="accordion-panel-:r0:"
                  aria-expanded="false"
                  class="chakra-accordion__button css-uttm9k"
                  data-index="0"
                  id="accordion-button-:r0:"
                  type="button"
                >
                  <div
                    class="css-0"
                  >
                    First Element
                  </div>
                  <svg
                    aria-hidden="true"
                    class="chakra-icon chakra-accordion__icon css-186l2rg"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </h2>
              <div
                class="chakra-collapse"
                style="overflow: hidden; display: none; opacity: 0; height: 0px;"
              >
                <div
                  aria-labelledby="accordion-button-:r0:"
                  class="chakra-accordion__panel css-mxa4qn"
                  id="accordion-panel-:r0:"
                  role="region"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    Content in First Element
                  </p>
                </div>
              </div>
            </div>
            <div
              class="chakra-accordion__item css-17mg6aq"
            >
              <h2>
                <button
                  aria-controls="accordion-panel-:r1:"
                  aria-expanded="false"
                  class="chakra-accordion__button css-uttm9k"
                  data-index="1"
                  id="accordion-button-:r1:"
                  type="button"
                >
                  <div
                    class="css-0"
                  >
                    Second Element
                  </div>
                  <svg
                    aria-hidden="true"
                    class="chakra-icon chakra-accordion__icon css-186l2rg"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </h2>
              <div
                class="chakra-collapse"
                style="overflow: hidden; display: none; opacity: 0; height: 0px;"
              >
                <div
                  aria-labelledby="accordion-button-:r1:"
                  class="chakra-accordion__panel css-mxa4qn"
                  id="accordion-panel-:r1:"
                  role="region"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    Content in Second Element
                  </p>
                </div>
              </div>
            </div>
            <div
              class="chakra-accordion__item css-17mg6aq"
            >
              <h2>
                <button
                  aria-controls="accordion-panel-:r2:"
                  aria-expanded="false"
                  class="chakra-accordion__button css-uttm9k"
                  data-index="2"
                  id="accordion-button-:r2:"
                  type="button"
                >
                  <div
                    class="css-0"
                  >
                    Third Element
                  </div>
                  <svg
                    aria-hidden="true"
                    class="chakra-icon chakra-accordion__icon css-186l2rg"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </h2>
              <div
                class="chakra-collapse"
                style="overflow: hidden; display: none; opacity: 0; height: 0px;"
              >
                <div
                  aria-labelledby="accordion-button-:r2:"
                  class="chakra-accordion__panel css-mxa4qn"
                  id="accordion-panel-:r2:"
                  role="region"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    Content in Third Element
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <div
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-accordion css-1wbdjbz"
    >
      <div
        class="css-0"
      >
        <p
          class="css-0"
        >
          Accordion Title
        </p>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-svjswr"
      />
      <div
        class="css-0"
      >
        <div
          class="css-0"
        >
          <div
            class="chakra-accordion__item css-17mg6aq"
          >
            <h2>
              <button
                aria-controls="accordion-panel-:r0:"
                aria-expanded="false"
                class="chakra-accordion__button css-uttm9k"
                data-index="0"
                id="accordion-button-:r0:"
                type="button"
              >
                <div
                  class="css-0"
                >
                  First Element
                </div>
                <svg
                  aria-hidden="true"
                  class="chakra-icon chakra-accordion__icon css-186l2rg"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </h2>
            <div
              class="chakra-collapse"
              style="overflow: hidden; display: none; opacity: 0; height: 0px;"
            >
              <div
                aria-labelledby="accordion-button-:r0:"
                class="chakra-accordion__panel css-mxa4qn"
                id="accordion-panel-:r0:"
                role="region"
              >
                <p
                  class="chakra-text css-0"
                >
                  Content in First Element
                </p>
              </div>
            </div>
          </div>
          <div
            class="chakra-accordion__item css-17mg6aq"
          >
            <h2>
              <button
                aria-controls="accordion-panel-:r1:"
                aria-expanded="false"
                class="chakra-accordion__button css-uttm9k"
                data-index="1"
                id="accordion-button-:r1:"
                type="button"
              >
                <div
                  class="css-0"
                >
                  Second Element
                </div>
                <svg
                  aria-hidden="true"
                  class="chakra-icon chakra-accordion__icon css-186l2rg"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </h2>
            <div
              class="chakra-collapse"
              style="overflow: hidden; display: none; opacity: 0; height: 0px;"
            >
              <div
                aria-labelledby="accordion-button-:r1:"
                class="chakra-accordion__panel css-mxa4qn"
                id="accordion-panel-:r1:"
                role="region"
              >
                <p
                  class="chakra-text css-0"
                >
                  Content in Second Element
                </p>
              </div>
            </div>
          </div>
          <div
            class="chakra-accordion__item css-17mg6aq"
          >
            <h2>
              <button
                aria-controls="accordion-panel-:r2:"
                aria-expanded="false"
                class="chakra-accordion__button css-uttm9k"
                data-index="2"
                id="accordion-button-:r2:"
                type="button"
              >
                <div
                  class="css-0"
                >
                  Third Element
                </div>
                <svg
                  aria-hidden="true"
                  class="chakra-icon chakra-accordion__icon css-186l2rg"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </h2>
            <div
              class="chakra-collapse"
              style="overflow: hidden; display: none; opacity: 0; height: 0px;"
            >
              <div
                aria-labelledby="accordion-button-:r2:"
                class="chakra-accordion__panel css-mxa4qn"
                id="accordion-panel-:r2:"
                role="region"
              >
                <p
                  class="chakra-text css-0"
                >
                  Content in Third Element
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
